File: Basics/Prime Numbers/prime-numbers.py
def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def generate_primes_up_to_limit(limit):
    return [x for x in range(2, limit + 1) if is_prime(x)]

limit = 50
primes = generate_primes_up_to_limit(limit)
print(f"Prime numbers up to {limit}: {primes}")

File: Basics/Prime Numbers/prime-numbers.rs
fn generate_primes_up_to(limit: usize) -> Vec<usize> {
    let mut primes = Vec::new();
    let mut is_prime = vec![true; limit + 1];

    for num in 2..=limit {
        if is_prime[num] {
            primes.push(num);

            let mut multiple = 2 * num;
            while multiple <= limit {
                is_prime[multiple] = false;
                multiple += num;
            }
        }
    }

    primes
}

fn main() {
    let limit = 30; // Set your desired limit here
    let primes = generate_primes_up_to(limit);

    println!("Prime numbers up to {}: {:?}", limit, primes);
}